### DISCUSSION TITLE
Jail creation arguments (jailmaker create --ARGS)

### DISCUSSION THREAD

**Discussion by spusuf** [Jip-Hop/jailmaker#40] [Ideas]
Would it be possible to add arguments to the initial `jlmkr create` script to reduce the amount of time spent creating jails, as well as being able to recreate/share jail configs easier.

The way I'd prefer to do it is by copy pasting a one line creation script such as:
```
jailmaker create --name='JAILNAME' --release=debian12 --macvlan --intel  --nvidia=0 --bind='/mnt/tank:/home/' --autostart --start  
``` 
And would return a confirmation such as `Create a debian12 jail named JAILNAME. Network set to macvlan. Intel GPU passed through. NVIDIA GPU not passed through. /mnt/tank on host system bound to /home/ . Autostart set to yes. Will start when created`

Where --intel --autostart and --start are shorthand for intel=1 or intel=y.
--release=default should use recommended distro. 

Even rudimentary functionality would be a HUGE time saver for me. I dont need confirmation or shorthands as the basic functionality is whats really required, but they would be nice.

**Comment by Jip-Hop** [comment]
I like the idea. The [bash version of jailmaker](https://gist.github.com/Jip-Hop/4704ba4aa87c99f342b2846ed7885a5d/5ac6fa849f094e82bace079c586678115850cd22) worked a bit more like how you describe. Initially it was only configurable with environment variables. Later I added the 'guided install wizard' because of feedback that it would be easier to use. Then I rewrote jailmaker in python and added additional features, but I only ported the guided install.

I personally have only 1 jail and almost never need to create another one. So jailmaker does (more) than what I need. Adding the feature you suggest would take some refactoring (I once started working on it). But I'll leave this one up to the community until (if ever) I need this functionally myself.

**Reply by spusuf** [reply]
I'm creating multiple jails a day (and destroying them) for testing, development, and verification, so any form of streamlining would be appreciated.

**Comment by Jip-Hop** [comment]
I'm locking this discussion as I've opened dedicated issues to track feature requests. Please see [Non-interactive CLI](https://github.com/Jip-Hop/jailmaker/issues/40).
